frontend/public/components/monitoring/targets.tsx

```diff
@@ -0,0 +1,365 @@
+  loaded: boolean;
+  loadError: string;
+  targets: Target[];
+};
+
+const REDUX_ID = 'monitoringTargets';
+
+const getRowProps = (target: Target) => ({ id: target.scrapeUrl, title: target.lastError });
+
+const List: React.FC<ListProps> = ({ loaded, loadError, targets }) => {
+  const { t } = useTranslation();
+
+  const filters = useSelector(({ k8s }: RootState) => k8s.getIn([REDUX_ID, 'filters']));
+
+  const Header = () => [
+    {
+      title: t('public~Endpoint'),
+      sortField: 'scrapeUrl',
+      transforms: [sortable],
+      props: { className: tableClasses[0] },
+    },
+    {
+      title: t('public~Monitor'),
+      sortField: 'labels.service',
+      transforms: [sortable],
+      props: { className: tableClasses[1] },
+    },
+    {
+      title: t('public~Status'),
+      sortField: 'health',
+      transforms: [sortable],
+      props: { className: tableClasses[2] },
+    },
+    {
+      title: t('public~Namespace'),
+      sortField: 'labels.namespace',
+      transforms: [sortable],
+      props: { className: tableClasses[3] },
+    },
+    {
+      title: t('public~Last Scrape'),
+      sortField: 'lastScrape',
+      transforms: [sortable],
+      props: { className: tableClasses[4] },
+    },
+    {
+      title: t('public~Scrape Duration'),
+      sortFunc: 'targetScrapeDuration',
+      transforms: [sortable],
+      props: { className: tableClasses[5] },
+    },
+  ];
+
+  const rowFilters: RowFilter[] = [
+    {
+      filterGroupName: t('public~Status'),
+      items: [
+        { id: 'up', title: t('public~Up') },
+        { id: 'down', title: t('public~Down') },
+      ],
+      reducer: (target: Target) => target?.health,
+      type: 'observe-target-health',
+    },
+    {
+      filterGroupName: t('public~Source'),
+      items: [
+        { id: AlertSource.Platform, title: t('public~Platform') },
+        { id: AlertSource.User, title: t('public~User') },
+      ],
+      reducer: (target: Target) => targetSource(target),
+      type: 'observe-target-source',
+    },
+  ];
+
+  return (
+    <>
+      <Helmet>
+        <title>{t('public~Metrics targets')}</title>
+      </Helmet>
+      <div className="co-m-nav-title">
+        <h1 className="co-m-pane__heading">{t('public~Metrics Targets')}</h1>
+      </div>
+      <div className="co-m-pane__body">
+        {loadError && (
+          <Alert
+            className="co-alert"
+            title={t('public~Error loading latest targets data')}
+            variant="danger"
+          >
+            {loadError}
+          </Alert>
+        )}
+        <FilterToolbar
+          data={targets}
+          labelFilter="observe-target-labels"
+          labelPath="labels"
+          nameFilterPlaceholder={t('public~Search by endpoint or monitor...')}
+          reduxIDs={[REDUX_ID]}
+          rowFilters={rowFilters}
+          textFilter="observe-target-text"
+        />
+        <div className="row">
+          <div className="col-xs-12">
+            <Table
```

Comment:
Don't redo this, but in general new code should use the updated table components from #8829.